coretex/cli/modules/utils.py

```diff
@@ -17,20 +17,80 @@
 
 from typing import List, Any, Optional, Callable
 from functools import wraps
+from importlib.metadata import version as getLibraryVersion
 
 import sys
+import logging
 
 from py3nvml import py3nvml
 
 import click
+import requests
 
+from . import ui
 from ...utils.process import command
 
 
 def updateLib() -> None:
     command([sys.executable, "-m", "pip", "install", "--no-cache-dir", "--upgrade", "coretex"], ignoreStdout = True, ignoreStderr = True)
 
 
+def fetchLatestVersion() -> Optional[str]:
+    url = "https://pypi.org/pypi/coretex/json"
+    response = requests.get(url)
+
+    if response.ok:
+        data = response.json()
+
+        infoDict = data.get("info")
+        if not isinstance(infoDict, dict):
+            logging.getLogger("cli").debug("Value of json field of key \"info\" in \"data\" dictionary is not of expected type (dict).")
+            return None
+
+        version = infoDict.get("version")
+        if not isinstance(version, str):
+            logging.getLogger("cli").debug("Value of json field of key \"version\" in \"info\" dictionary is not of expected type (str).")
+            return None
+
+        return version
+
+    logging.getLogger("cli").debug(f"Failed to fetch version of coretex library. Response code: {response.status_code}")
+    return None
+
+
+def isVersionStrValid(version: str) -> bool:
+    parts = version.split('.')
+
+    if len(parts) != 3:
+        return False
+
+    return all(part.isdigit() for part in parts)
+
+
+def checkLibVersion() -> None:
+    current = getLibraryVersion("coretex")
+    latest = fetchLatestVersion()
+
+    if not isinstance(latest, str):
+        logging.getLogger("cli").debug(f"Invalid type of \"latest\" var {type(current)}. Expected \"string\".")
+        return
+
+    if not isVersionStrValid(current):
+        logging.getLogger("cli").debug("Current version returned by importlib library is not valid.")
+        return
+
+    if not isVersionStrValid(latest):
+        logging.getLogger("cli").debug("Latest version extracted from pypi website is not valid.")
+        return
+
+    majorCurrent, minorCurrent, patchCurrent = map(int, current.split('.'))
+    majorLatest, minorLatest, patchLatest = map(int, latest.split('.'))
```

Comment:
What if the int conversion fails?

Reply:
But won't isVersionStrValid() function check if function is valid for conversion, in what case will conversion to int fails if version is validated with isVersionStrValid() function before?